<a href="https://colab.research.google.com/github/Mat9408/Atividades_CursoAnalistaDeDados_EBAC/blob/main/M%C3%B3dulo_22_Tarefa_(3)_(1).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!wget -q 'https://raw.githubusercontent.com/andre-marcos-perez/ebac-course-utils/develop/dataset/traffic.csv' -O traffic.csv

In [ ]:
import numpy as np

data = None

with open(file='traffic.csv', mode='r', encoding='utf8') as fp:

  fp.readline()
  data = fp.read()

day = 14
incidents = list()
incident_by_day = dict()

for timebox in data.split(sep='\n'):
  timebox_data = timebox.split(sep=';')
  incident = np.sum(np.array([int(i) for i in timebox_data[1: len(timebox_data)-1]]))
  incidents.append(incident)

  try:
    half_hour = int(timebox_data[0])
    if half_hour == 27:
      incident_by_day[day] = incidents
      day = day + 1
      incidents = list()
  except ValueError:
    continue

for day in incident_by_day:
  print(f'{day}: {incident_by_day[day]}')

14: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 5, 2, 0, 0, 0, 0, 0, 1, 0, 0, 0, 2, 2, 3]
15: [3, 0, 0, 0, 2, 4, 2, 1, 3, 0, 2, 0, 2, 3, 2, 2, 4, 7, 6, 3, 0, 1, 1, 5, 4, 6, 0]
16: [1, 1, 2, 4, 4, 5, 3, 2, 4, 2, 1, 1, 3, 6, 2, 0, 2, 8, 4, 6, 4, 1, 2, 7, 15, 4, 0]
17: [4, 1, 1, 0, 0, 0, 2, 3, 2, 7, 2, 1, 7, 3, 6, 4, 3, 2, 2, 3, 2, 5, 2, 0, 1, 4, 15]
18: [10, 1, 1, 5, 5, 7, 3, 4, 3, 1, 2, 3, 2, 2, 6, 2, 2, 4, 2, 3, 0, 3, 3, 2, 3, 6, 0]


###Métricas

In [ ]:
media_14 = np.mean(incident_by_day[14])
print(media_14)

desvio_14 = np.std(incident_by_day[14])
print(desvio_14)

0.5925925925925926
1.1944085553034889


In [ ]:
media_15 = np.mean(incident_by_day[15])
print(media_15)

desvio_15 = np.std(incident_by_day[15])
print(media_15)

2.3333333333333335
2.3333333333333335


In [ ]:
media_16 = np.mean(incident_by_day[16])
print(media_16)

desvio_16 = np.std(incident_by_day[16])
print(desvio_16)

3.4814814814814814
3.047407767514698


In [ ]:
media_17 = np.mean(incident_by_day[17])
print(media_17)

desvio_17 = np.std(incident_by_day[17])
print(desvio_17)

3.037037037037037
3.0487578707046925


In [ ]:
media_18 = np.mean(incident_by_day[18])
print(media_18)

desvio_18 = np.std(incident_by_day[18])
print(desvio_18)

3.1481481481481484
2.1892617342417258


###Interpretação

- A maior média de acidentes a cada meia hora é do dia 16. O valor registrado foi de 3.48 ocorrências.

- A menor variação de acidentes a cada meia hora foi apontada no dia 14. O valor alcançado foi de 1.19 ocorrências.